code/game/objects/items/devices/paicard.dm

```diff
@@ -323,4 +323,27 @@
 	if(pai && pai.client && !pai.canmove)
 		var/rendered = "<span class='message'>[text]</span>"
 		pai.show_message(rendered, 2)
-	..()
\ No newline at end of file
+	..()
+
+/obj/item/device/paicard/dropped(mob/user)
+
+	///When an object is put into a container, drop fires twice.
+	//once with it on the floor, and then once in the container
+	//This conditional allows us to ignore that first one.

```

Comment:
Could have just removed it and kept the second check at that rate. Then only that condition forces action.